### ISSUE TITLE
Remove undertow app-server

### ISSUE THREAD

**Issue by rmartinc** [keycloak/keycloak#30347]
### Description

Right now it's possible to test the saml adapters using the `-Papp-server-undertow` which needs the app-server built for undertow (folder `testsuite/integration-arquillian/servers/app-server/undertow`). But this is never tested and it fails right now. After some investigation the failures are due two reasons:

1. The current version of undertow is jakarta based and the tests are still prepared for javax.
2. After fixed the previous point is fixed there are several individual failures (we need to fix them test by test).

In summary, SAML adapter have not been tested inside undertow for a looooooong time. So I think it's better to just test what we support (wildfly/eap) and remove the undertow app server completely. This way there will be not possibility of testing with app-server undertow.
